novawallet/Modules/Vote/Governance/TinderGov/Helpers/ReferendumObservableSource.swift

```diff
@@ -0,0 +1,128 @@
+import SoraFoundation
+import Operation_iOS
+
+protocol ReferendumsSourceObserver: AnyObject {
+    func didReceive(_ changes: [DataProviderChange<ReferendumLocal>])
+}
+
+protocol ReferendumObservableSourceProtocol {
+    func observe(_ target: ReferendumsSourceObserver)
+}
+
+protocol ReferendumObservableStoreProtocol: ReferendumObservableSourceProtocol {
+    func save(referendums: [ReferendumLocal])
+    func remove(with ids: [ReferendumIdLocal])
+}
+
+protocol WeakWrappersClearing: AnyObject {
+    var observers: [WeakWrapper] { get set }
+}
+
+extension WeakWrappersClearing {
+    func clearEmptyWrappers() {
+        observers.removeAll { $0.target == nil }
+    }
+}
+
+final class ReferendumObservableSource {
+    var observers: [WeakWrapper] = []
+    private var referendums: [ReferendumIdLocal: ReferendumLocal] = [:]
+
+    private let mutex = NSLock()
+}
+
+// MARK: ReferendumObservableSourceProtocol
+
+extension ReferendumObservableSource: ReferendumObservableStoreProtocol, WeakWrappersClearing {
+    func save(referendums: [ReferendumLocal]) {
```

Comment:
We call save in the ReferendumsInteractor and never call remove. So will have still have removed referendums in the list.